src/contour/TerminalSession.cpp

```diff
@@ -651,23 +653,65 @@ void TerminalSession::pasteFromClipboard(unsigned count, bool strip)
         sessionLog()("pasteFromClipboard: mime data contains {} formats.", md->formats().size());
         for (int i = 0; i < md->formats().size(); ++i)
             sessionLog()("pasteFromClipboard[{}]: {}\n", i, md->formats().at(i).toStdString());
-        string const text = strip_if(normalize_crlf(clipboard->text(QClipboard::Clipboard)), strip);
-        if (text.empty())
+
+        auto const text = clipboard->text(QClipboard::Clipboard);
+
+        // 1 MB hard limit
+        if (text.size() > 1024 * 1024)
```

Comment:
I think this is definitely a good self-protection to have. Pasting that much into `stdin` most likely smells like either abuse, or somebody not knowing that this will freeze your session. We can see if we need to tweak the number in the future.